api/src/main/java/org/commonjava/maven/galley/model/SimpleLocation.java

```diff
@@ -31,15 +31,17 @@
 
     private final boolean allowReleases;
 
+    private final boolean allowDeletion;
+
     private final String uri;
 
     private final Map<String, Object> attributes = new HashMap<String, Object>();
 
     private final String name;
 
     public SimpleLocation( final String name, final String uri, final boolean allowSnapshots, final boolean allowReleases,
- final boolean allowsStoring, final boolean allowPublishing,
-                           final boolean allowDownloading )
+                           final boolean allowsStoring, final boolean allowPublishing, final boolean allowDownloading,
+                           final boolean allowDeletion )
```

Comment:
I'm not sure if any other projects are using this class, so suggest to retain former constructor and add this as new one. Just set the allowDeletion to true as default for the former one.